STM32Cube/Tasks/sdmmc.c

```diff
@@ -0,0 +1,187 @@
+#include "sdmmc.h"
+#include <stdio.h>
+
+UART_HandleTypeDef *sdmmcHuart;
+
+const char *logsPath = "/LOGS"; // DO NOT CHANGE
+char logFileName[500];
+
+void print(const char *formatter, ...) {
+	va_list args;
+	va_start(args, formatter);
+
+	char output[1024] = {0};
+
+	vsnprintf(output, 1024, formatter, args);
+	va_end(args);
+
+	HAL_UART_Transmit(sdmmcHuart, (uint8_t *)output, strlen(output), 10);
+}
+
+FRESULT list_dir(char *path) {
+	print("READING DIR '%s'\n", path);
+	FRESULT res;
+	DIR dir;
+	FILINFO fno;
+	int nfile, ndir;
+
+	res = f_opendir(&dir, path); /* Open the directory */
+	if (res == FR_OK) {
+		nfile = ndir = 0;
+		for (;;) {
+			res = f_readdir(&dir, &fno); /* Read a directory item */
+			if (res != FR_OK || fno.fname[0] == 0)
+				break;                    /* Error or end of dir */
+			if (fno.fattrib & AM_DIR) { /* Directory */
+				print("   <DIR>   %s\n", fno.fname);
+				ndir++;
+			} else {
+				print("%10u %s\n", fno.fsize, fno.fname);
+				nfile++;
+			}
+		}
+		f_closedir(&dir);
+		print("%d dirs, %d files.\n", ndir, nfile);
+	} else {
+		print("Failed to open \"%s\". (%u)\n", path, res);
+	}
+
+	return res;
+}
+
+int computeFolderSize(const char *path) {
+	DIR dir;
+	FILINFO fno;
+	int nfile = 0;
+
+	if (f_opendir(&dir, path) == FR_OK) {
+		while (f_readdir(&dir, &fno) == FR_OK && fno.fname[0]) {
+			if (!(fno.fattrib & AM_DIR))
+				++nfile;
+		}
+		f_closedir(&dir);
+	}
+
+	return nfile;
+}
+
+void initUniqueLogFileName(const char *logsPath) {
```

Comment:
i guess `static` causes the linker to ignore multiple definition errors for both instances? either way, why do you have this here and in log.c? I would prefer if as much of the FATFS stuff as possible lived in this source file, and you pass the logfile name back to the calling task.